lib/plenario2/actions/data_set_constraint_actions.ex

```diff
@@ -6,15 +6,33 @@ defmodule Plenario2.Actions.DataSetConstraintActions do
 
   import Ecto.Query
 
+  import Plenario2.Guards, only: [is_id: 1]
+
   alias Plenario2.Changesets.DataSetConstraintChangesets
   alias Plenario2.Schemas.{DataSetConstraint, Meta}
   alias Plenario2.Repo
 
+  @typedoc """
+  Parameter is an ID attribute
+  """
+  @type id :: String.t | integer
+
+  @typedoc """
+  Returns a tuple of :ok, DataSetConstraint or :error, Ecto.Changeset
+  """
+  @type ok_constr :: {:ok, DataSetConstraint} | {:error, Ecto.Changeset.T}
+
   @doc """
   Creates a new instance of a DataSetConstraint.
   """
-  @spec create(meta_id :: integer, field_names :: [String.t]) :: {:ok, %DataSetConstraint{} | :error, Ecto.Changeset.t}
-  def create(meta_id, field_names) do
+  @spec create(meta :: Meta | id, field_names :: [String.t]) :: ok_constr
+  def create(meta, field_names) do
+    meta_id =
+      case is_id(meta) do
+        true -> meta
+        false -> meta.id
+      end
```

Comment:
Could be more succinct  as two heads of the same function.

```elixir
def create(meta, fields) when !is_id(meta), do: create(meta.id, fields)
def create(meta_id, fields) do
  # work with id
end
```

Both ways you've done it are readable, but using function guards over case is more elixir(ific? tastic?) I believe